### PR TITLE
Add `deploy` integration test

### PR DESCRIPTION
**Pull request by serinamarie**
This PR adds an integration test for `.deploy()` for versions 2.18.0 and greater. It also pins the existing integration tests for `.apply(), `Deployment`, and agents to <= 2.19.0.

New integration test in action:
```bash
--------------------------------------- deploy.py ----------------------------------------
Created work pool 'test-pool'!
To start a worker for this work pool, run:
        prefect worker start --pool test-pool
Looks like you're deploying to a process work pool. If you're creating a 
deployment for local development, calling `.serve` on your flow is a great way 
to get started. See the documentation for more information: 
https://docs.prefect.io/latest/concepts/flows/#serving-a-flow.  Set 
`ignore_warnings=True` to suppress this message.
Successfully created/updated all deployments!
                 Deployments                 
┏━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━┳━━━━━━━━━┓
┃ Name                  ┃ Status  ┃ Details ┃
┡━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━╇━━━━━━━━━┩
│ hello/demo-deployment │ applied │         │
└───────────────────────┴─────────┴─────────┘
To execute flow runs from this deployment, start a worker in a separate terminal
that pulls work from the 'test-pool' work pool:
        $ prefect worker start --pool 'test-pool'
To schedule a run for this deployment, use the following command:
        $ prefect deployment run 'hello/demo-deployment'
You can also run your flow via the Prefect UI: http://127.0.0.1:4200/deployments/deployment/37bb4689-3b9e-4140-856d-d8b611539f76
21:04:05.210 | INFO    | prefect.engine - Created flow run 'miniature-fulmar' for flow 'hello'
21:04:05.210 | INFO    | Flow run 'miniature-fulmar' - View at http://127.0.0.1:4200/flow-runs/flow-run/d3e6192d-6952-4154-bd12-3287344371f7
21:04:05.246 | INFO    | Flow run 'miniature-fulmar' - Hello demo-deployment!
21:04:05.410 | INFO    | Flow run 'miniature-fulmar' - Finished in state Completed()
Deleted work pool 'test-pool'
------------------------------------------------------------------------------------------
```

### CODE REVIEW

File: .github/workflows/integration-tests.yaml [hunk 1]
```diff
@@ -10,6 +10,7 @@ on:
       - ui/**
       - .nvmrc
       - Dockerfile
+      - flows/
```
Comment (serinamarie) [inline .github/workflows/integration-tests.yaml:13]: I think we should run these tests when we make changes to the `flows/` unless there is any opposition

File: flows/prefect2_only/deployment_storage.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+
+async def read_flow_run(flow_run_id):
+    async with prefect.get_client() as client:
+        return await client.read_flow_run(flow_run_id)
+
+
+def main():
+    if Version(prefect.__version__) <= Version("2.19.0"):
+        from prefect.deployments import Deployment
```
Comment (serinamarie) [inline flows/prefect2_only/deployment_storage.py:50]: This moves `Deployment` import behind a version check as we won't have `Deployment` after 2.19.0

File: flows/prefect2_only/deployment_run.py [hunk 3]
```diff
@@ -45,25 +44,28 @@ async def read_flow_run(flow_run_id):
...
-        env=env,
-        timeout=30,
-        stdout=sys.stdout,
-        stderr=sys.stderr,
-    )
-
-    flow_run = anyio.run(read_flow_run, flow_run.id)
-    assert flow_run.state.is_completed(), flow_run.state
+    if Version(prefect.__version__) <= Version("2.19.0"):
+        from prefect.deployments import Deployment
```
Comment (serinamarie) [inline flows/prefect2_only/deployment_run.py:48]: This moves `Deployment` import behind a version check as we won't have `Deployment` after 2.19.0. No other changes

File: flows/prefect2_only/deployment_apply.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+async def apply_deployment(deployment):
+    async with prefect.get_client() as client:
+        flow_id = await client.create_flow_from_name(deployment.flow_name)
+        await client.create_deployment(flow_id=flow_id, name=deployment.name)
+
+
+if __name__ == "__main__":
+    if Version(prefect.__version__) <= Version("2.19.0"):
+        from prefect.deployments import Deployment
```
Comment (serinamarie) [inline flows/prefect2_only/deployment_apply.py:21]: This moves `Deployment` import behind a version check as we won't have `Deployment` after 2.19.0. No other changes

File: flows/deploy.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+                    source="https://github.com/PrefectHQ/prefect-recipes.git",
+                    entrypoint="flows-starter/hello.py:hello",
+                )
+
+                await flow_instance.deploy(
+                    name="demo-deployment",
+                    work_pool_name="test-pool",
+                )
+
+                flow_run_state = flow_instance._run("demo-deployment")
```
Comment (desertaxle) [inline flows/deploy.py:43]: I think it'd make more sense to use `run_deployment` followed by a call to `prefect flow-run execute` to smoke test an end-to-end run.

**Review by desertaxle — COMMENTED** [review]
With the new `prefect2_only` subdirectory, I think `scripts/run-integration-flows.py` needs to be updated to run flows in subdirectories as well.

File: flows/deploy.py (hunk 5 above)
Comment (serinamarie) [reply flows/deploy.py]: That makes sense, I've applied in [2a6086d](https://github.com/PrefectHQ/prefect/pull/13392/commits/2a6086d38b82e03fb7739833f94351a78d38593b)

**Comment by serinamarie** [comment]
> With the new `prefect2_only` subdirectory, I think `scripts/run-integration-flows.py` needs to be updated to run flows in subdirectories as well.

@desertaxle That makes sense, I've applied that in [6badeb3](https://github.com/PrefectHQ/prefect/pull/13392/commits/6badeb3171fe62d8e43c3b0c88d0e6b889f2d5e5)

**Review by desertaxle — APPROVED** [review]

File: scripts/run-integration-flows.py [hunk 6]
```diff
@@ -38,7 +38,7 @@ def run_flows(search_path: Union[str, Path]):
     if server_version:
         print(f"and server version: {server_version}")
 
-    for file in sorted(Path(search_path).glob("*.py")):
+    for file in sorted(Path(search_path).glob("**/*.py")):
```
Comment (serinamarie) [inline scripts/run-integration-flows.py:41]: Ensures we run on any file in the `flows/` directory.

**Pull request merged**